### DISCUSSION TITLE
Is there any way to set the limit to zero?

### DISCUSSION THREAD

**Discussion by wolfogre** [meilisearch/product#730] [Feedback & Feature Proposal]
Thanks for the amazing meilisearch project!

I encountered a problem while using it. I'm wondering if I missed anything.

I just want to get the total number of the search, no need to return any result to speed up. But zero limit will be reset to the default limit, and magic numbers like -1 are not allowed.

I encountered this issue in the Gitea project: go-gitea/gitea#29990, https://github.com/go-gitea/gitea/pull/29990/commits/f886a75c822ace8320553ca4db2f3c7ae581afd6

Currently, I have to set it to 1, then clear the result before returning. It works, but I wonder if there's a better way to do it, since it seems Elasticsearch works well with zero limit.

**Comment by macraig** [comment]
Hi @wolfogre , thanks for the feedback! You didn't miss anything, the `limit` parameter only accepts integers > 0 currently. I will move this discussion to the feature requests repo.

**Reply by dureuill** [reply]
To clarify, this looks like a limitation of the [go SDK](https://github.com/meilisearch/meilisearch-go), specifically.

A raw `curl` call to Meilisearch returns the expected:

```
❯ curl http://localhost:7700/indexes/test4/search -H 'Content-Type: application/json' --data-binary '{"sort": ["Info.ReviewNb:desc"], "limit": 0}'
{"hits":[],"query":"","processingTimeMs":0,"limit":0,"offset":0,"estimatedTotalHits":7}%
```

The behavior of the Go SDK was introduced by [this commit](https://github.com/meilisearch/meilisearch-go/commit/59a8827ae920033ca20e4ff11f8bfb65e37c7383)

This bit of code was added to https://github.com/meilisearch/meilisearch-go/issues/4 because previously not passing a limit to the Go SDK would result in the limit being 0.

It appears to be a [limit of the Go language](https://github.com/meilisearch/meilisearch-go/issues/4#issuecomment-605969844) that uninitialized values are 0, but I'm not well versed enough in Go to know if there is a better solution.

I would recommend opening an issue against the Go SDK.
